tests/lib/prepare.sh

```diff
@@ -955,6 +955,19 @@ EOF
         snap download --basename=pc --channel="20/$GADGET_CHANNEL" pc
         test -e pc.snap
         unsquashfs -d pc-gadget pc.snap
+
+        # change the schema to DOS if that env var is set
+        if [ "${CORE_GADGET_SCHEMA:-}" != "" ]; then
+            if os.query is-core20 || os.query is-core22; then
+                # to use DOS on UC20+, we cannot have ubuntu-save otherwise we 
+                # have too many partitions for DOS, so remove ubuntu-save from
+                # the gadget.yaml since we can't use encryption on GCE anyways
+                # and the tests using DOS are mainly about gadget asset updates
+                # anyways
+                sed -i "/name: ubuntu-save/,+4 d" pc-gadget/meta/gadget.yaml
+            fi
+            sed -i -e "s@bootloader: grub@schema: $CORE_GADGET_SCHEMA\n    bootloader: grub@g" pc-gadget/meta/gadget.yaml
```

Comment:
Hm I'm confused by how this is used. The bits modified here are for the reflash magic, which is done only once in prepare and then the system boots into this image. So if I'm not mistaken, when the same host executes both variants of the test, the other one will fail, as the image contains a gadget for the first variant.

Reply:
I see, we need to check this. Also this block of changes is already inside an `if os.query is-core20` condition, which I suspect was not the desired intention.

Reply:
So, I've run the core tests on core20 (`spread -debug google:ubuntu-core-20-64:tests/core/`) and they all pass (except the netconfig one, but that's for other reasons), **while having this whole part of the diff removed**. First the mbr variant of the test was run, then the gpt.

Reply:
Hi, what I understand 

is that in his if [ "${CORE_GADGET_SCHEMA:-}" != "" ]; then
CORE_GADGET_SCHEMA will the empty first time when the image is built
so the code in the if won't be executed but in the tests the prepare image won't be call anymore so that part won't be executed at all

I think to do what he was trying to do, this should be a nested test
so in nested you build the image using the test env vars